### PR TITLE
Update copy for workspace sharing setting

### PR DESCRIPTION
**Pull request by loujaybee**
Small update for copy text on workspace sharing. 

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@loujaybee: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/dashboard/src/teams/TeamSettings.tsx [hunk 1]
```diff
@@ -150,7 +150,7 @@ export default function TeamSettingsPage() {
                         title={<span>Workspace Sharing</span>}
                         desc={
                             <span>
-                                Allow organization members to share running workspaces outside the organization.
+                                Allow workspaces creаted within an Organization to share the workspace with any authenticated user.
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamSettings.tsx:153]: Slight adjustment as the current copy reads like this setting still allows you to share with users within the Organization.

**Review by easyCZ — APPROVED** [review]

File: components/dashboard/src/teams/TeamSettings.tsx (hunk 1 above)
Comment (gtsiolis) [reply components/dashboard/src/teams/TeamSettings.tsx:153]: Found this change a bit confusing. Some thoughts and questions:
1. Does this mean workspaces for projects within the organization?
2. Do we need to use a title case for the organization?
3. Do we need to explicitly mention that only authenticated users can open a shared workspace or any workspace? 

Cc @loujaybee @easyCZ

Comment (loujaybee) [reply components/dashboard/src/teams/TeamSettings.tsx:153]: > Does this mean workspaces for projects within the organization?

Good point, we need to clarify, I assumed so, but we should be explicit. I will also consider [adding to the docs](https://github.com/gitpod-io/website/pull/3548). 

> Do we need to use a title case for the organization?

Ah yes, probably should have 😬 

> Do we need to explicitly mention that only authenticated users can open a shared workspace or any workspace?

The challenge here is that it's only workspaces for that Organization. In Dedicated context, saying "only workspaces created in your Organization" sounds odd as there will be only one org. But not in Public SaaS 🤔 If you have some suggestions to improve further, let's add them?

**Pull request merged**